exporter/elasticsearchexporter/model.go

```diff
@@ -385,7 +352,109 @@ func numberToValue(dp pmetric.NumberDataPoint) (pcommon.Value, error) {
 	return pcommon.Value{}, errInvalidNumberDataPoint
 }
 
-func (m *encodeModel) encodeSpan(resource pcommon.Resource, span ptrace.Span, scope pcommon.InstrumentationScope) ([]byte, error) {
+func (m *encodeModel) encodeResourceOTelMode(document *objmodel.Document, resource pcommon.Resource, resourceSchemaURL string) {
+	resourceMapVal := pcommon.NewValueMap()
+	resourceMap := resourceMapVal.Map()
+	if resourceSchemaURL != "" {
```

Comment:
It is expected that this empty string check will cause the new test added in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34471 to fail. Let's merge https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34471 first, then rebase this PR and fix the test.